packages/rulesets/src/spectral/az-arm.ts

```diff
@@ -73,18 +75,52 @@ const ruleset: any = {
         function: longRunningResponseStatusCodeArm,
       },
     },
+
     // RPC Code: RPC-Async-V1-02
-    ProvisioningStateSpecified: {
-      description: 'A LRO PUT and PATCH operations response schema must have "ProvisioningState" property specified.',
+    //PUT
+    ProvisioningStateSpecifiedForLROPut: {
+      description:
+        'A LRO PUT operation\'s response schema must have "ProvisioningState" property specified for the 200 and 201 status codes.',
+      message: "{{error}}",
+      severity: "error",
+      resolved: true,
+      formats: [oas2],
+      given: ["$[paths,'x-ms-paths'].*[put][?(@property === 'x-ms-long-running-operation' && @ === true)]^"],
+      then: {
+        function: provisioningStateSpecifiedForLROPut,
+      },
```

Comment:
Instead of using a custom function here, you can use the core function [`truthy`](https://meta.stoplight.io/docs/spectral/cb95cf0d26b83-core-functions#truthy).
```suggestion
      given: ["$[paths,'x-ms-paths'].*[put][?(@property === 'x-ms-long-running-operation' && @ === true)]^.responses[200,201].schema"],
      then: {
        field: "provisioningState",
        function: truthy,
      },
```

Reply:
Yeah!
But we need to fetch/collect the properties right?

Reply:
@AkhilaIlla I think it should work since the `given` already looks for `'x-ms-long-running-operation' === true` and with my change it will just grab the schema of 200 and 201 responses. Maybe try swapping in my suggestion and see if your tests still work. You might need to change the expected message in the tests too.

Reply:
It looks like the only property you are checking in the custom function is `provisioningState`.